eng/testing/SetIcuMode.cmd

```diff
@@ -0,0 +1,40 @@
+:: This script is intended for our ICU tests
+:: the reason why we need to do this is because we build the tests
+:: in one version of Windows and then send the tests to helix to
+:: multiple Windows SKUs, including SKUs that doesn't contain ICU,
+:: since the globalization mode is calculated very early in the process
+:: we can't enable running ICU at all in this SKUs and then skip the tests
+:: at runtime, because xunit will start and then blow up when trying to load ICU.
+:: So with this script, we check if icu.dll is not present on the system, then we
+:: fallback to run on NLS mode.
+
+@echo off
+setlocal ENABLEEXTENSIONS
+setlocal ENABLEDELAYEDEXPANSION
+
+if EXIST %WINDIR%\system32\icu.dll (
+    echo ICU exists in system, running tests with ICU.
+    goto done
+)
+
+echo icu was not found, so running the tests using NLS.
```

Comment:
Can we tell from test logs or otherwise that ICU tests actually run? (ie., if there was a bug in this script that meant it always fell back to NLS at all times, how would we know)

Reply:
If this script fails on platforms where ICU tests should've run there are two tests that check for that and will fail. 

If the script fails to fallback to NLS on platforms that ICU is not available it will crash.

All this info we should be able to figure out with the console logs as the script is echoing the status of it.

Reply:
These are the two tests that would fail if we're not running on ICU mode.

https://github.com/dotnet/runtime/blob/dd7738a6b4fa583f5627044b069ca1699476844f/src/libraries/System.Globalization/tests/IcuTests/IcuTests.cs#L16